src/routes/(app)/product/[id]/+page.server.ts

```diff
@@ -113,36 +113,25 @@ export const load = async ({ params, locals }) => {
 		.find({ productId: params.id })
 		.sort({ createdAt: 1 })
 		.toArray();
-	const freeProductSubscriptions = await collections.paidSubscriptions
+	const subscriptions = await collections.paidSubscriptions
 		.find({
 			...userQuery(userIdentifier(locals)),
-			paidUntil: { $gt: new Date() },
-			[`freeProductsById.${product._id}`]: { $exists: true }
+			paidUntil: { $gt: new Date() }
 		})
-		.sort({ [`freeProductsById.${product._id}.available`]: -1 })
-		.limit(1)
 		.toArray();
-	const subscriptions = freeProductSubscriptions
-		? []
-		: await collections.paidSubscriptions
-				.find({
-					...userQuery(userIdentifier(locals)),
-					paidUntil: { $gt: new Date() }
-				})
-				.toArray();
+	const freeProductSubscriptions = subscriptions.filter(
+		(s) => (s.freeProductsById?.[product._id]?.available ?? 0) > 0
+	);
+	const percentageSubscriptions = freeProductSubscriptions ? [] : subscriptions;
```

Comment:
`freeProductSubscriptions` is always defined, you should use  `freeProductSubscriptions.length` or even compute the sum directly here.

By the way is there a reason for this line? In my opinion we still want to show the discounted price (so if they go beyond their free products, they already know the prices they will get)